exist-core/src/main/java/org/exist/xquery/functions/fn/ExtCollection.java

```diff
@@ -215,22 +208,26 @@ private URI asUri(final String path) throws XPathException {
     }
 
     private Sequence toSequence(final DocumentSet docs) throws XPathException {
-        final NodeSet result = new NewArrayNodeSet();
+        final Sequence result = new ValueSequence();
         final LockManager lockManager = context.getBroker().getBrokerPool().getLockManager();
         for (final Iterator<DocumentImpl> i = docs.getDocumentIterator(); i.hasNext(); ) {
             final DocumentImpl doc = i.next();
 
-            ManagedDocumentLock dlock = null;
-            try {
-                if (!context.inProtectedMode()) {
-                    dlock = lockManager.acquireDocumentReadLock(doc.getURI());
-                }
-                result.add(new NodeProxy(doc));
-            } catch (final LockException e) {
-                throw new XPathException(this, ErrorCodes.FODC0002, e);
-            } finally {
-                if (dlock != null) {
-                    dlock.close();
+            // filter out binary documents, fn:collection should only return XML documents
+            if (doc.getResourceType() == DocumentImpl.XML_FILE) {
```

Comment:
Since no code will be executed if this check fails I would reverse this check and return immediately.
`if (doc.getResourceType() == DocumentImpl.XML_FILE) { return; }`
The benefit is more readable, less indented code. Less context to keep in mind.

Reply:
I see your point, the flip-side though is that it will increase the cyclic complexity. Also I find `continue` to be a non-obvious thing

Reply:
I personally am used to write and read return-early style code. I also doubt that cyclomatic complexity increases. On the other hand this is not important enough to block this PR.

Reply:
I had a second look and must admit the complexity would raise by 1 because of the extra control-flow element.